### ISSUE TITLE
(This is not an issue with the code) Need help on adding layers before retinanet layers

### ISSUE THREAD

**Issue by kbeznakparmatonic** [fizyr/keras-retinanet#367]
Hi,
          I am trying to combine retinanet with few layers to enhance the object detection and I want to add a model before retinanet. I have tried the following option to combine both the model.

Model(inputs=model1.input, outputs=retinanet(model1.output))

But the issue is, once I get to the line where we decide the Loss function for outputs, the optimizer gives error.
The code snippet is from train.py after the creation of the model.
```python
training_model.compile(
        loss={
            'regression'    : losses.smooth_l1(),
            'classification': losses.focal()
            #'retinanet-bbox'    : losses.smooth_l1(),
            #'retinanet-bbox': losses.focal()
        },
        optimizer=keras.optimizers.adam(lr=1e-5, clipnorm=0.001)
    )
```
The error is that the output layer names have been changed to 'retinanet-bbox' instead of regression and classification and etc. I am trying to retain the code structure as much as possible and want to combine my model without disturbing this situation. Can you please help me with this ?

**Comment by hgaiser** [comment]
Hmm I've never tried this, but you should define the model like this:

```python
Model(inputs=model1.input, outputs=retinanet(model1.output).outputs)
```

**Comment by kbeznakparmatonic** [comment]
I have tried that, but retinanet(model1.output) will give you list and we an't use outputs on that.

**Comment by hgaiser** [comment]
The `retinanet` function will give you a `keras.models.Model`. What is the list you're referring to?

**Comment by hgaiser** [comment]
Is your issue resolved?

**Comment by rgsl888** [comment]
My question was is there a better way to combine my model "A"  with retina-net such that A will provide the input to the retina-net. And as I have already mentioned this change "Model(inputs=model1.input, outputs=retinanet(model1.output).outputs) " will modify the outputs being used by the optimizer.

**Comment by hgaiser** [comment]
Sorry for the late reply.

I think it should be possible to combine the outputs of both your model and retinanet? You would get something like this:
```python
Model(inputs=model1.input, outputs=model1.outputs + retinanet(model1.outputs).outputs)
```
That should combine both your outputs and retinanet's.

**Comment by rgsl888** [comment]
Thank you, that helped a lot.

**Comment by hgaiser** [comment]
Okay I'll close this issue then.
